contracts/ExampleTokens.cdc

```diff
@@ -0,0 +1,255 @@
+    ///
+    pub fun createEmptyVault(tokenID: UInt64): @TokenVault {
+        return <-create TokenVault(tokenID: tokenID, balance: 0.0)
+    }
+
+    pub resource Administrator {
+
+        /// createNewMinter
+        ///
+        /// Function that creates and returns a new minter resource
+        /// Minter can mint an allowance of tokens of the specified ID
+        ///
+        pub fun createNewMinter(tokenID: UInt64, allowedAmount: UFix64): @Minter {
+            emit MinterCreated(tokenID: tokenID, allowedAmount: allowedAmount)
+            return <-create Minter(tokenID: tokenID, allowedAmount: allowedAmount)
+        }
+
+        /// createNewBurner
+        ///
+        /// Function that creates and returns a new burner resource
+        ///
+        pub fun createNewBurner(): @Burner {
+            emit BurnerCreated()
+            return <-create Burner()
+        }
+    }
+
+    /// Minter
+    ///
+    /// Resource object that token admin accounts can hold to mint new tokens.
+    ///
+    pub resource Minter {
+
+        /// The amount of tokens that the minter is allowed to mint
+        pub var allowedAmount: UFix64
+        pub var tokenID: UInt64
+
+        /// mintTokens
+        ///
+        /// Function that mints new tokens, adds them to the total supply,
+        /// and returns them to the calling context.
+        ///
+        pub fun mintTokens(amount: UFix64): @ExampleTokens.TokenVault {
+            pre {
+                amount > 0.0: "Amount minted must be greater than zero"
+                amount <= self.allowedAmount: "Amount minted must be less than the allowed amount"
+            }
+            ExampleTokens.totalSupplyByID[self.tokenID] = ExampleTokens.totalSupplyByID[self.tokenID]! + amount
+            self.allowedAmount = self.allowedAmount - amount
+            emit TokensMinted(tokenID: self.tokenID, amount: amount)
+            return <-create TokenVault(tokenID: self.tokenID, balance: amount)
+        }
+
+        init(tokenID: UInt64, allowedAmount: UFix64) {
+            self.allowedAmount = allowedAmount
+            self.tokenID = tokenID
+        }
+    }
+
+    /// Burner
+    ///
+    /// Resource object that token admin accounts can hold to burn tokens.
+    ///
+    pub resource Burner {
+
+        /// burnTokens
+        ///
+        /// Function that destroys a Vault instance, effectively burning the tokens.
+        ///
+        /// Note: the burned tokens are automatically subtracted from the
+        /// total supply in the Vault destructor.
+        ///
+        pub fun burnTokens(from: @FungibleTokens.TokenVault) {
+            let vault <- from as! @ExampleTokens.TokenVault
+            let amount = vault.balance
+            let tokenID = vault.tokenID
+            destroy vault
+            emit TokensBurned(tokenID: tokenID, amount: amount)
+        }
+    }
+
+    pub resource Collection: FungibleTokens.CollectionPublic, FungibleTokens.CollectionPrivate {
```

Comment:
I love the idea of having collections for multiple fungible tokens. I think we'll definitely want to integrate something like this in the standard

Reply:
I basically tried to create something that follows existing patterns from FT/NFT interfaces as there has been very little talk of updating the existing standard and the longer they are left in their current form the greater the friction breaking changes will introduce.